### ISSUE TITLE
[1.13-rc4] `docker swarm init` gets stuck in "pending" instead of error about "address already in use"

### ISSUE THREAD

**Issue by dnephin** [docker/docker#29885]
**Description**

In 1.13-rc4, `docker swarm init` does not exit. It gets stuck without printing an error. If I ctrl-c the client and try again, I get an error saying: 
```
Error response from daemon: This node is already part of a swarm
```

```
$ docker info
...
Swarm: pending
 Is Manager: false
...
```

**Steps to reproduce the issue:**
1. `nc -l 0.0.0.0 4040`
2. `docker swarm init --listen-addr 0.0.0.0:4040`

**Describe the results you received:**

The command hangs and does not init the swarm. The swarm is left in a "pending" state.

**Describe the results you expected:**

This error:
```
Error response from daemon: failed to listen on remote API address: listen tcp 0.0.0.0:4040: bind: address already in use
```

**Additional information you deem important:**

Daemon log:
```
DEBU[0013] Calling POST /v1.25/swarm/init
DEBU[0013] form data: {"AdvertiseAddr":"","AutoLockManagers":false,"ForceNewCluster":false,"ListenAddr":"0.0.0.0:4040","Spec":{"CAConfig":{},"Dispatcher":{},"EncryptionConfig":{"AutoLockManagers":false},"Orchestration":{},"Raft":{"ElectionTick":0,"HeartbeatTick":0},"TaskDefaults":{}}}
DEBU[0013] generated CA key and certificate              module=node
DEBU[0013] no node credentials found in: /tmp/dockerdev/swarm/certificates/swarm-node.crt  error="open /tmp/dockerdev/swarm/certificates/swarm-node.key: no such file or directory" module="node/tls"
DEBU[0013] loaded TLS certificate                        module="node/tls" node.id=xv3w2j6d91w2tfnuxzka51um6
DEBU[0013] issued new TLS certificate                    module="node/tls" node.id=xv3w2j6d91w2tfnuxzka51um6 node.role=swarm-manager
DEBU[0013] new node credentials generated: /tmp/dockerdev/swarm/certificates/swarm-node.crt  module="node/tls" node.id=xv3w2j6d91w2tfnuxzka51um6 node.role=swarm-manager
DEBU[0013] next certificate renewal scheduled            module="node/tls" node.id=xv3w2j6d91w2tfnuxzka51um6 node.role=swarm-manager time=2017-03-15 22:58:00.000000061 -0400 EDT
```

**Output of `docker version`:**

```
Client:
 Version:      1.13.0-rc4
 API version:  1.25
 Go version:   go1.7.4
 Git commit:   88862e7
 Built:        Wed Jan  4 16:40:23 2017
 OS/Arch:      linux/amd64

Server:
 Version:      1.13.0-rc4
 API version:  1.25 (minimum version 1.12)
 Go version:   go1.7.4
 Git commit:   88862e7
 Built:        Wed Jan  4 16:40:23 2017
 OS/Arch:      linux/amd64
 Experimental: false
```

**Output of `docker info`:**

```
Containers: 0
 Running: 0
 Paused: 0
 Stopped: 0
Images: 0
Server Version: 1.13.0-rc4
Storage Driver: aufs
 Root Dir: /tmp/dockerdev/aufs
 Backing Filesystem: extfs
 Dirs: 0
 Dirperm1 Supported: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins: 
 Volume: local
 Network: bridge host macvlan null overlay
Swarm: pending
 NodeID: xv3w2j6d91w2tfnuxzka51um6
 Is Manager: false
 Node Address: 10.88.111.3
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 03e5862ec0d8d3b3f750e19fca3ee367e13c090e
runc version: 51371867a01c467f08af739783b8beafc154c4d7
init version: 949e6fa
Security Options:
 apparmor
 seccomp
  Profile: default
Kernel Version: 3.19.0-33-generic
Operating System: Ubuntu 14.04.4 LTS
OSType: linux
Architecture: x86_64
CPUs: 8
Total Memory: 15.61 GiB
Name: dnephin-desktop
ID: LYJN:DJWY:EK46:S27B:PF2U:6X43:JUKD:2CTD:LXFA:KHX3:5UXP:W7BS
Docker Root Dir: /tmp/dockerdev
Debug Mode (client): false
Debug Mode (server): true
 File Descriptors: 16
 Goroutines: 37
 System Time: 2017-01-04T12:06:11.732118643-05:00
 EventsListeners: 0
Username: dnephin
Registry: https://index.docker.io/v1/
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false
```

**Comment by tonistiigi** [comment]
This seems to happen because after https://github.com/docker/swarmkit/commit/6379ef019e1158238cf40b7d40cb6c43c11d1b8d#diff-380e79adccbaf1276bf6c91cc551cd7f when one of the swarm components crashes it doesn't bring down the whole node anymore. Unless I'm missing something this looks more dangerous than just "address already in use" case and may cause any swarmkit error to not bubble up to docker.

Note that this doesn't appear on master because this commit is only vendored in `1.13.x` branch.

@dongluochen @aaronlehmann @LK4D4

**Comment by dnephin** [comment]
Reopening until the swarmkit change is vendored.

**Comment by aaronlehmann** [comment]
Should be fixed now that swarmkit was vendored in #29926.
